eth_account/messages.py

```diff
@@ -145,6 +145,84 @@ def encode_structured_data(
     :param text: the message as a series of unicode characters (a normal Py3 str)
     :returns: The EIP-191 encoded message, ready for signing
 
+
+    Usage Notes:
+     - An EIP712 message consists of 4 top-level keys: ``types``, ``primaryType``,
+       ``domain``, and ``message``. All 4 must be present to encode properly.
+     - The key ``EIP712Domain`` must be present within ``types``.
+     - The `type` of a field may be:
+        - a Solidity `atomic` type: ``bytes32``, ``uint256``, ``address``, etc.
+        - a Solidity `dynamic` type: ``string`` or ``bytes``
```

Comment:
Can we just say `atomic`/`dynamic` types to make it more generic?

Reply:
It's a distinction I had in my head from writing tests, but doesn't have any bearing for in this function. Generified.